### PR TITLE
Part of #19125: Migrate component to TS: Tag

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- This PR is a part of  #19125 
Migrated the components from JS to TS for `Tag`
Filepath: `ui\components\component-library\tag`
It extends the `BoxProps` so the controls have increased

### Before

image
image

### After

image
image

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Review by georgewrmarshall — APPROVED** [review]
Hey @dhruvv173, thanks for your contribution, you hit it out of the park! LGTM 🚀

File: ui/components/component-library/tag/tag.types.ts
Comment (georgewrmarshall) [inline ui/components/component-library/tag/tag.types.ts:2]: Ah this will fix the error you are getting
```suggestion
import type { BoxProps } from '../../ui/box/box.d';
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Ah sorry I missed the error. One small update!

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, thanks for your contribution, you hit it out of the park! LGTM 🚀 

Sorry I didn't leave a message for review as a dependency test was failing. However, you've gone out of your way to help me out with it. Thanks a lot!

I'll make the changes and submit the PR. Thanks once again!

**Comment by dhruvv173** [comment]
Hey @georgewrmarshall I have made the changes as suggested, could you please review them?
Thanks!

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, yes updating the deprecated consts for enums would be great! Thanks

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, yes updating the deprecated consts for enums in the Tag component would be great! Thanks

Hey @georgewrmarshall, sorry for the delayed response but by the time I started working on updating `consts -> enums` the open PR related to it was already merged. So I just had to make changes in the `Tag` component file. 
Could you please review it?
Thanks

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Hey @dhruvv173, in my final checks I found some things that could be updated
- deprecated storybook `ComponentMeta` tags
- PropType issue in storybook

Screenshot 2023-05-20 at 11 54 13 AM
Screenshot 2023-05-20 at 11 54 20 AM

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, in my final checks I found some things that could be updated
> 
> * deprecated storybook `ComponentMeta` tags
> * PropType issue in storybook


hey @georgewrmarshall, the deprecated component issue was due to change in the storybook version to `v7` for which my branch wasn't updated. However, its fixed now. 
image


- I looked into the PropType issue and found out that the control should be passed in as an array and not an object and thats when the changes are reflected correctly. Does that need to be changed? If yes, I'd like to do it if you could guide me.
Thanks
image
image

**Comment by dhruvv173** [comment]
hey @georgewrmarshall, just an update on this issue.
I was going through the already migrated components and faced the same PropType issue for the extended `Box` props.
The props defined in the `.types.ts` file work fine but its the extended ones which cause this issue.
image

Works fine when passed as an array
image

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, Thanks for your time and effort in creating this pull request. We have identified an issue related to the typing of the `Box` component, specifically with the polymorphic `as` prop. To address this issue, we have an open ticket at https://github.com/MetaMask/metamask-extension/issues/19239 and a draft pull request at https://github.com/MetaMask/metamask-extension/pull/19363 that needs to be merged before we can proceed with your PR. Thanks for your patience

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, Thanks for your time and effort in creating this pull request. We have identified an issue related to the typing of the `Box` component, specifically with the polymorphic `as` prop. To address this issue, we have an open ticket at #19239 and a draft pull request at #19363 that needs to be merged before we can proceed with your PR. Thanks for your patience

hey, can this be worked on now?

**Comment by georgewrmarshall** [comment]
Hey @dhruvv173, one more blocker that I'm working on now https://github.com/MetaMask/metamask-extension/pull/19572

**Comment by dhruvv173** [comment]
> Hey @dhruvv173, one more blocker that I'm working on now https://github.com/MetaMask/metamask-extension/pull/19572

Apologies for missing that part out:/
I'll keep an eye on that PR and start working once it's merged, thankyou for bringing it to my attention!

**Comment by georgewrmarshall** [comment]
Blocked by https://github.com/MetaMask/metamask-extension/pull/19572

**Comment by dhruvv173** [comment]
hello @georgewrmarshall , made changes with a clean commit history and resolved conflicts [here](https://github.com/MetaMask/metamask-extension/pull/20086)

**Pull request closed without merging**
